services/accounts/accounts.go

```diff
@@ -37,3 +48,131 @@ func (api *API) GetAccounts(ctx context.Context) ([]accounts.Account, error) {
 func (api *API) DeleteAccount(ctx context.Context, address types.Address) error {
 	return api.db.DeleteAccount(address)
 }
+
+func (api *API) AddAccountWatch(ctx context.Context, address string, name string, color string) error {
+	account := accounts.Account{
+		Address: types.Address(common.HexToAddress(address)),
+		Type:    "watch",
+		Name:    name,
+		Color:   color,
+	}
+	return api.SaveAccounts(ctx, []accounts.Account{account})
+}
+
+func (api *API) AddAccountWithMnemonic(
+	ctx context.Context,
+	mnemonic string,
+	address string,
+	password string,
+	name string,
+	color string,
+) error {
+	mnemonicNoExtraSpaces := strings.Join(strings.Fields(mnemonic), " ")
+
+	_, err := api.manager.VerifyAccountPassword(api.config.KeyStoreDir, address, password)
```

Comment:
I think we should be able to avoid having to pass `address` as a parameter. I think `wallet_root_address` in `settings` table contains it.  (same comment applies to `AddAccountWithPrivateKey`, and `GenerateAccount`)